src/libraries/System.Formats.Tar/tests/TarTestsBase.cs

```diff
@@ -611,5 +612,37 @@ public static IEnumerable<object[]> GetPaxAndGnuTestCaseNames()
                 yield return new object[] { name };
             }
         }
+
+        public static bool CanExtractWithTarTool = CanExtractWithTarTool_Method();
+
+        private static bool CanExtractWithTarTool_Method()
+        {
+            using Process tarToolProcess = new Process();
+            tarToolProcess.StartInfo.FileName = "tar"; // location varies: find it on the PATH.
+            tarToolProcess.StartInfo.Arguments = "--help";
+
+            tarToolProcess.StartInfo.RedirectStandardOutput = true;
+            tarToolProcess.Start();
```

Comment:
this will fail if the test is run on a version of Windows prior to when they added 'tar'. do we have a "find on path" method or existing code we can use? or, we could catch whatever the exception is when the file does not exist.

Reply:
According to this: https://docs.microsoft.com/en-us/virtualization/community/team-blog/2017/20171219-tar-and-curl-come-to-windows

> Beginning in Insider Build 17063, we’re introducing two command-line tools to the Windows toolchain: curl and bsdtar

I assume they're talking about Windows 11. Interestingly, my Windows 10 machine also has tar available, so I think the document is incomplete.

Reply:
On Windows 10, executing `tar -h` on PowerShell gets me this:
```
bsdtar 3.5.2 - libarchive 3.5.2 zlib/1.2.5.f-ipp
```
And on Windows 11:

```
bsdtar 3.5.2 - libarchive 3.5.2 zlib/1.2.5.f-ipp bz2lib/1.0.6 
```

Reply:
[Decoder](https://en.wikipedia.org/wiki/List_of_Microsoft_Windows_versions) .. this is Windows 10 RS4, in 2018.

Reply:
Ah good to know. The article's date is `04/25/2022`, that's why I was assuming Win11.

Reply:
Also note we [support ](https://github.com/dotnet/core/blob/main/release-notes/7.0/supported-os.md) Windows 10 from RS1 (1607)

Reply:
@danmoseley if it fails, it will just skip the test, something like this:
```
    Discovering: System.Formats.Tar.Tests (method display = ClassAndMethod, method display options = None)
    Discovered:  System.Formats.Tar.Tests (found 1 of 578 test case)
    Starting:    System.Formats.Tar.Tests (parallel test collections = on, max threads = 12)
      System.Formats.Tar.Tests.TarFile_CreateFromDirectory_File_Tests.ResultingArchive_CanBeExtractedByOtherTools [SKIP]
        Condition(s) not met: "CanExtractWithTarTool (TargetInvocationException)"
    Finished:    System.Formats.Tar.Tests
  === TEST EXECUTION SUMMARY ===
     System.Formats.Tar.Tests  Total: 1, Errors: 0, Failed: 0, Skipped: 1, Time: 0.172s
```

Reply:
I will wrap the code in a try-catch, just in case.

Reply:
Now I wonder, if that's how xUnit handles exceptions on expressions used for `ConditionalFact/TheoryAttribute`s, does it mean that we can end up with tests no longer running and we will never notice?
cc @dotnet/area-infrastructure-libraries